pkg/cfn/manager/nodegroup.go

```diff
@@ -215,6 +215,9 @@ func (c *StackCollection) GetUnmanagedNodeGroupAutoScalingGroupName(ctx context.
 	if err != nil {
 		return "", err
 	}
+	if res.StackResourceDetail.PhysicalResourceId == nil {
```

Comment:
I'm not sure what status we should consider for returning an error: CREATE_FAILED & DELETE_COMPLETE ?
I think it's still safer to check the PhysicalResourceId before dereferencing it, even if we check the status before.

Reply:
I'm not saying it isn't safer to check the `PhysicalResourceId`, but rather that it's cleaner to act upon a stack status of `CREATE_FAILED` or `DELETE_COMPLETE`. The PhysicalResourceId should never be nil for a stack status of `CREATE_COMPLETE`.

Reply:
@vflaux, there are other failure values like `ROLLBACK_FAILED`, `UPDATE_ROLLBACK_FAILED` etc., where it's unclear if the `PhysicalResourceId` will be present.  Instead of covering all the possible failure or success status values, I think we should just remove the status check and use the `PhysicalResourceId == nil` check. I'm retracting my original suggestion.

Reply:
Ok. I removed the check on the `ResourceStatus` and the associated test.